transcom/mymove
# B 18085 identify create values 7 12 integration Kickback

## B-18085 Kickback

## Summary

MilMove needs to provide the PPM Closeout Counselor with a complete view/listing of PPM Costing Calculations upon PPM Closeout. This PR adds a null check for SIT items and checks if there has been an aoa requested. This should stop the tool from crashing when those fields are null.

This PR handles values 7-12:
1. Gross Incentive
2. Government Constructive Cost(GCC)
3. Advance Operating Allowance(AOA)
4. Remaining Incentive
5. Linehaul/short haul price
6.  Linehaul/short haul fsc

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Use any PPM move and/or use testharness "MoveWithPPMShipmentReadyForFinalCloseoutWitSIT"
2. Log in as a service counselor and review and submit move
4. Log in as user for that move
5. Upload documents
(This should set the PPM move status to  "NEEDS_PAYMENT_APPROVAL")
8. In postman go to the Get ppmCloseout endpoint and use that move's PPM ID as the "ppmShipmentID"
9. Verify that fields 7-12 are populated

To find ppmID using locator in DBeaver:
```
  select m.locator as move_code
         , m.id as move_id
         , ms.id as mto_shipment_id
         , ps.id as ppm_shipment_id
         , ms.shipment_type
    from moves m 
         inner join mto_shipments ms
         on m.id = ms.move_id
         inner join ppm_shipments ps
         on ms.id = ps.shipment_id
   where 1 = 1
     and m.locator = 'your_locator'
     and ms.shipment_type = 'PPM'
  order by m."locator"
           , ms.id 
           , ms.shipment_type 
```
           

## Screenshots
_**no advance requested and no SIT**_
<img width="704" alt="Screenshot 2024-01-30 at 2 24 43 PM" src="https://github.com/transcom/mymove/assets/146969726/e58b33e2-81ec-49f4-945a-fe16b72b2591">

**_advanced recieved and has SIT_**
<img width="708" alt="Screenshot 2024-01-30 at 2 24 27 PM" src="https://github.com/transcom/mymove/assets/146969726/c41916a9-cc30-424d-b8fa-9890a02e3fe2">
